caluma/caluma_core/management/commands/cleanup_history.py

```diff
@@ -3,9 +3,11 @@
 
 # Instances which are created at run-time, referencing config-time models
 # eg. Answers are created at run-time by users and referencing Question
+# NOTE: This checks indirectly if not-nullable referenced objects exist.
+# Theoretically it doesn't matter which field to check for, but localized fields seem flaky.
 RELATED_HISTORICAL_MODELS = {
     "Answer": {"question__type__isnull": True},
-    "Document": {"form__name__isnull": True},
+    "Document": {"form__is_published__isnull": True},
     "DynamicOption": {"question__type__isnull": True},
 }
```

Comment:
Wouldn't checking for the related `pk` make more sense? Or am I missing something?

Reply:
The point of this command is clean up broken data, so required ForeignKeys that don't exist anymore. This wouldn't be possible in normal usage and is not allowed by Django, but can only happen by mishandling fixtures or manipulating the db directly.

Trying to access the pks directly will give an error, hence the indirection via a field of the related table.

Reply:
:+1: